server/routes/api/protein.py

```diff
@@ -25,10 +30,292 @@
 @bp.route('/<path:dcid>')
 def get_node(dcid):
     """Returns data given a protein node."""
-    response = dc_service.fetch_data('/internal/bio', {
+    response = dc.fetch_data('/internal/bio', {
         'dcid': dcid,
     },
-                                     compress=False,
-                                     post=False,
-                                     has_payload=False)
+                             compress=False,
+                             post=False,
+                             has_payload=False)
     return response
+
+
+def id(id_or_dcid):
+    '''
+    Delete 'bio/' prefix from a biomedical DCID, if it is present.
+    E.g. 'bio/P53_HUMAN' --> 'P53_HUMAN'
+    '''
+    # .replace replaces all instances, but no biomedical DCID will contain more than once instance of 'bio/'
+    return id_or_dcid.replace(BIO_DCID_PREFIX, '')
+
+
+def dcid(id_or_dcid):
+    '''
+    Add 'bio/' prefix to an id if it does not already start with 'bio/'.
+    E.g. 'P53_HUMAN' --> 'bio/P53_HUMAN'
+    '''
+    if id_or_dcid.startswith(BIO_DCID_PREFIX):
+        return id_or_dcid
+    return BIO_DCID_PREFIX + id_or_dcid
+
+
+def node(protein_id_or_dcid, depth):
+    '''
+    Returns node object.
+    Protein dcids are of the form 'bio/{name}_{species}' (e.g. 'bio/P53_HUMAN')
+    '''
+    node_id = id(protein_id_or_dcid)
+    protein, species = node_id.split('_')
+    return {
+        'id': node_id,
+        'name': protein,
+        'species': species,
+        'depth': depth,
+    }
+
+
+def extract_intactmi(score_dcids):
+    '''
+    Takes list of interaction score DCIDs and numerically parses and returns the first instance
+    of an IntactMi score, or the default interaction score if none are found.
+    
+    TODO: add example
+    '''
+    for score_dcid in score_dcids:
+        if score_dcid.startswith("IntactMiScore"):
+            return float(score_dcid.replace("IntactMiScore", ""))
+    return DEFAULT_INTERACTION_SCORE
+
+
+def interactors(interaction_id_or_dcid):
+    '''
+    Parse and return ids of two interactors from interaction DCID of the form 'bio/{protein1 id}_{protein2 id}'
+    E.g. 'bio/P53_HUMAN_CBP_HUMAN' --> 'P53_HUMAN', 'CBP_HUMAN'
+    '''
+    interaction_id = id(interaction_id_or_dcid)
+    protein1, species1, protein2, species2 = interaction_id.split('_')
+    return f'{protein1}_{species1}', f'{protein2}_{species2}'
+
+
+def target(interaction_id_or_dcid, source_id_or_dcid):
```

Comment:
looks like this method is only ever called with an interaction_dcid and a source_id so maybe update the arguments

Reply:
sounds good!